.github/workflows/test.yml

```diff
@@ -119,10 +119,14 @@ jobs:
         env:
           SYNAPSE_ACCESS_TOKEN: ${{ secrets.SYNAPSE_ACCESS_TOKEN }}
           SERVICE_ACCOUNT_CREDS: ${{ secrets.SERVICE_ACCOUNT_CREDS }}
+        # run: >
+        #   source .venv/bin/activate;
+        #   pytest --durations=0 --cov-report=term --cov-report=html:htmlcov --cov-report=xml:coverage.xml --cov=schematic/
+        #   -m "table_operations" -n auto --log-level=DEBUG -v
         run: >
           source .venv/bin/activate;
           pytest --durations=0 --cov-report=term --cov-report=html:htmlcov --cov-report=xml:coverage.xml --cov=schematic/
-          -m "not (rule_benchmark or table_operations)" --reruns 2 -n auto
```

Comment:
As we discussed, locally, these tests run really fast, but for some reason on github actions, they take a long time... Do you have a difference in time?  Perhaps it's also github actions being weird

Reply:
I ran this without the xdist concurrency at it took around 5 hours:
https://github.com/Sage-Bionetworks/schematic/actions/runs/10623727416/job/29450709775?pr=1482

With xdist at a best case scenario this is going to be running in 2.5 hours. Locally these tests run in around 10 minutes.

I ran this in debug mode so we have more context on what is happening and when. I saw that in-between HTTP calls, there are long delays when running in GH. For example, in this log there are 22 seconds where nothing appears to be occurring:

```
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 receive_response_headers.started request=<Request [b'PUT']>
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 receive_response_headers.complete return_value=(b'HTTP/1.1', 201, b'', [(b'Date', b'Thu, 29 Aug 2024 18:04:17 GMT'), (b'Content-Type', b'application/json;charset=UTF-8'), (b'Content-Length', b'152'), (b'Connection', b'keep-alive'), (b'Server', b'34 f6e676 27, 1647 57 c616 4796f6 e6 3712 0295. f6 5702 66f657 e6 4602 f657 27 02 3756, 3627 5647e2.'), (b'X-XSS-Protection', b'1;mode=block'), (b'X-Content-Type-Options', b'nosniff'), (b'Access-Control-Allow-Origin', b'*'), (b'Strict-Transport-Security', b'max-age=31536000'), (b'Vary', b'Accept-Encoding,User-Agent'), (b'Content-Encoding', b'gzip'), (b'X-Frame-Options', b'SAMEORIGIN')])
Thu, 29 Aug 2024 18:04:18 GMT INFO     httpx:_client.py:1773 HTTP Request: PUT https://file-prod.prod.sagebase.org/file/v1/file/multipart/54232414/complete "HTTP/1.1 201 "
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 receive_response_body.started request=<Request [b'PUT']>
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 receive_response_body.complete
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 response_closed.started
Thu, 29 Aug 2024 18:04:18 GMT DEBUG    httpcore.http11:_trace.py:85 response_closed.complete
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 close.started
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 close.complete
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 close.started
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 close.complete
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.started host='repo-prod.prod.sagebase.org' port=443 local_address=None timeout=70 socket_options=None
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.complete return_value=<httpcore._backends.anyio.AnyIOStream object at 0x7f199f1c9ca0>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 start_tls.started ssl_context=<ssl.SSLContext object at 0x7f199f116ac0> server_hostname='repo-prod.prod.sagebase.org' timeout=70
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.complete return_value=<httpcore._backends.anyio.AnyIOStream object at 0x7f199f1e3af0>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 start_tls.started ssl_context=<ssl.SSLContext object at 0x7f199f116ac0> server_hostname='repo-prod.prod.sagebase.org' timeout=70
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.complete return_value=<httpcore._backends.anyio.AnyIOStream object at 0x7f199f1e3ee0>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 start_tls.started ssl_context=<ssl.SSLContext object at 0x7f199f116ac0> server_hostname='repo-prod.prod.sagebase.org' timeout=70
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 connect_tcp.complete return_value=<httpcore._backends.anyio.AnyIOStream object at 0x7f199f1c9e80>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 start_tls.started ssl_context=<ssl.SSLContext object at 0x7f199f116ac0> server_hostname='repo-prod.prod.sagebase.org' timeout=70
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.connection:_trace.py:85 start_tls.complete return_value=<httpcore._backends.anyio.AnyIOStream object at 0x7f199ee97040>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.http11:_trace.py:85 send_request_headers.started request=<Request [b'POST']>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.http11:_trace.py:85 send_request_headers.complete
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.http11:_trace.py:85 send_request_body.started request=<Request [b'POST']>
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.http11:_trace.py:85 send_request_body.complete
Thu, 29 Aug 2024 18:04:40 GMT DEBUG    httpcore.http11:_trace.py:85 receive_response_headers.started request=<Request [b'POST']>
```


https://github.com/Sage-Bionetworks/schematic/actions/runs/10620230339/job/29439597224

Without collecting the telemetry data here I am not sure how to diagnosis this further.


I am going to push 1 more commit to this branch before I swap it to ready for review that turns testing back on for all tests. We can look at the numbers at that point and determine if we want to run these tests during normal integration test runs.